// bson/ext/index.js
var bson = null;

try {
    console.log("Platform : "+process.platform);
    console.log("Process.arch : "+process.arch);
    console.log("aiguillage : "+(process.platform == "win32" && process.arch == "ia32"));
	// Load the precompiled win32 binary
	if(process.platform == "win32" && process.arch == "x64") {
	  bson = require('./win32/x64/bson');  
        alert("x64");
	} else if(process.platform == "win32" && process.arch == "ia32") {
        console.log("ceci is require le bson de win32 en ia2 version");
	  bson = require('bson');  
       console.log("charge rep courant");
	} else {
        alert("release");
	  bson = require('../build/Release/bson');  
	}	
} catch(err) {
	// Attempt to load the release bson version
	try {
		bson = require('../build/Release/bson');
        console.log("total ALERT !!!!!");
	} catch (err) {
         console.log("total ALERT !!!!!\n"+err);
		console.error("Failed to load c++ bson extension, using pure JS version");
		bson = require('../lib/bson/bson');
	}
}

exports.BSON = bson.BSON;
exports.Long = require('../lib/bson/long').Long;
exports.ObjectID = require('../lib/bson/objectid').ObjectID;
exports.DBRef = require('../lib/bson/db_ref').DBRef;
exports.Code = require('../lib/bson/code').Code;
exports.Timestamp = require('../lib/bson/timestamp').Timestamp;
exports.Binary = require('../lib/bson/binary').Binary;
exports.Double = require('../lib/bson/double').Double;
exports.MaxKey = require('../lib/bson/max_key').MaxKey;
exports.MinKey = require('../lib/bson/min_key').MinKey;
exports.Symbol = require('../lib/bson/symbol').Symbol;

// Just add constants tot he Native BSON parser
exports.BSON.BSON_BINARY_SUBTYPE_DEFAULT = 0;
exports.BSON.BSON_BINARY_SUBTYPE_FUNCTION = 1;
exports.BSON.BSON_BINARY_SUBTYPE_BYTE_ARRAY = 2;
exports.BSON.BSON_BINARY_SUBTYPE_UUID = 3;
exports.BSON.BSON_BINARY_SUBTYPE_MD5 = 4;
exports.BSON.BSON_BINARY_SUBTYPE_USER_DEFINED = 128;
